sympy/stats/tests/test_stochastic_process.py

```diff
@@ -194,3 +195,55 @@ def test_BernoulliProcess():
     assert P(B[5] > 0, B[5]) == BernoulliDistribution(0.6, 0, 1)
     raises(ValueError, lambda: P(3))
     raises(ValueError, lambda: P(B[3] > 0, 3))
+
+def test_random_indexed_symbols():
+    t = symbols('t', positive=True, integer=True)
+    X, Y = BernoulliProcess("X", 0.4), PoissonProcess("Y", 3)
+    assert random_indexed_symbols(X[t] + Y(t)**3) == [X[t], Y(t)]
+    assert not random_indexed_symbols(t + t**2)
+
+def test_PoissonProcess():
+    X = PoissonProcess("X", 3)
+    assert X.state_space == Interval(0, oo)
+    assert X.index_set == Interval(0, oo)
+    assert X.lamda == 3
+
+    t, d, x, y = symbols('t d x y', positive=True)
+    assert isinstance(X(t), RandomIndexedSymbol)
+    raises(ValueError, lambda: PoissonProcess("X", -1))
+    raises(NotImplementedError, lambda: X[t])
+    raises(IndexError, lambda: X(-5))
+
+    assert X.joint_distribution(X(2), X(3)) == JointDistributionHandmade(Lambda((X(2), X(3)),
+                6**X(2)*9**X(3)*exp(-15)/(factorial(X(2))*factorial(X(3)))))
+
+    assert X.joint_distribution(4, 6) == JointDistributionHandmade(Lambda((X(4), X(6)),
+                12**X(4)*18**X(6)*exp(-30)/(factorial(X(4))*factorial(X(6)))))
+
+    assert P(X(t) < 1) == exp(-3*t)
+    assert P(Eq(X(t), 0), (t > 3) & (t <= 5)) == exp(-6) # exp(-2*lamda)
+    res = P(Eq(X(t), 1), (t > 3) & (t <= 4))
+    assert res == 3*exp(-3)
+    assert P(Eq(X(t), 1) & Eq(X(d), 1) & Eq(X(x), 1) & Eq(X(y), 1),
+    (t >= 0) & (t < 1) & (d >= 1) & (d < 2) &(x >= 2) & (x < 3) & (y >= 3) & (y < 4)) == res**4
+    assert P(Eq(X(t), 2) & Eq(X(d), 3), (t > 0) & (t <= 2) & (d > 1) & (d <= 4)) == \
+                Probability(Eq(X(d), 3) & Eq(X(t), 2), (d <= 4) & (t <= 2) & (d > 1))
+
+    raises(ValueError, lambda: P(Eq(X(t), 2) & Eq(X(d), 3), (t > 0) & (t <= 2) & (d > 1))) # no bound on d
+    assert P(Eq(X(t), 2), Eq(t, 3)) == 81*exp(-9)/2
```

Comment:
Do the following examples work?
```python
P(Not(Eq(X(t1), 2) && Ge(X(t2), 3)), Le(t1, 4) && Gt(t1, 2) && Le(t2, 8) && Gt(t2, 7))
P(Eq(X(t), 2) || Ne(X(t), 4), Le(t1, 4) && Gt(t1, 2))
```

Reply:
```python
>>> P(Not(Eq(X(t1), 2) & Ge(X(t2), 3)), Le(t1, 4) & Gt(t1, 2) & Le(t2, 8) & Gt(t2, 7))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/smit/Smitlunagariya/sympy/sympy/stats/rv.py", line 806, in probability
    return pspace(condition).probability(condition, given_condition, evaluate, **kwargs)
  File "/home/smit/Smitlunagariya/sympy/sympy/stats/stochastic_process.py", line 55, in probability
    return self.process.probability(condition, given_condition, evaluate, **kwargs)
  File "/home/smit/Smitlunagariya/sympy/sympy/stats/stochastic_process_types.py", line 1214, in probability
    return _probability_stoch('Poisson', condition, evaluate=evaluate, **kwargs)
  File "/home/smit/Smitlunagariya/sympy/sympy/stats/stochastic_process_types.py", line 1090, in _probability_
stoch
    result = pspace(new_condition).probability(new_condition, **kwargs)
  File "/home/smit/Smitlunagariya/sympy/sympy/stats/rv.py", line 421, in probability
    expr = condition.lhs - condition.rhs
AttributeError: 'Not' object has no attribute 'lhs'
>>> P(Eq(X(t1), 2) & Ge(X(t2), 3), Le(t1, 4) & Gt(t1, 2) & Le(t2, 8) & Gt(t2, 7))
18*(-17/2 + exp(3))*exp(-9)
>>> P(Eq(X(t1), 2) | Ne(X(t1), 4), Le(t1, 4) & Gt(t1, 2))
1 - 36*exp(-6)
```
So, for `Not` we need to handle as 1-P(...)

Reply:
After a small fix:
```python
>>> P(Not(Eq(X(t1), 2) & Ge(X(t2), 3)), Le(t1, 4) & Gt(t1, 2) & Le(t2, 8) & Gt(t2, 7))
-18*(-17/2 + exp(3))*exp(-9) + 1
```